src/main/java/com/ecommerce/ECommerceApp/model/Product.java

```diff
@@ -10,7 +12,7 @@ public class Product {
     private int productID;
     private String productName;
     private String productDescription;
-    private float productPrice;
+    private BigDecimal productPrice;
```

Comment:
Replaced float with BigDecimal for monetary values to prevent rounding issues in financial calculations, ensuring precision and reliability. As Martin Fowler discusses in ["Technical Debt"](https://martinfowler.com/bliki/TechnicalDebt.html), addressing data accuracy early on reduces technical debt and avoids potential issues down the line.